test/src/test/java/org/corfudb/integration/CorfuReplicationLargeTxIT.java

```diff
@@ -110,22 +114,38 @@ private void testTxChunking(int numEntriesToWrite,
         corfuStoreStandby.subscribeListener(standbyListener, LogReplicationMetadataManager.NAMESPACE,
                 LR_STATUS_STREAM_TAG);
 
-        // Calculate the expected total number of streaming updates across all
-        // tables
+        // Calculate the expected total number of streaming updates across all tables
         int totalStreamingUpdates =
             expectedStreamingUpdatesPerTable * mapNameToMapStandby.size();
 
+        // MAX_SNAPSHOT_ENTRIES_APPLIED = 1.  So entries to the ProtobufDescriptor table will be applied in a batch
+        // of 1.  Hence, the number of transactions made = number of entries received from the Source Cluster(1 TX
+        // per entry).
+        int numExpectedTxOnProtobufDescriptorTable = activeRuntime.getTableRegistry().getProtobufDescriptorTable().size();
+
+        // The number of entries in the protobuf descriptor table on Source(or numExpectedTxOnProtobufDescriptorTable)
+        // must not be equal to MAX_SNAPSHOT_ENTRIES_APPLIED.  Otherwise we cannot verify that it was applied in
+        // batches.
+        Assert.assertNotEquals(numExpectedTxOnProtobufDescriptorTable, MAX_SNAPSHOT_ENTRIES_APPLIED);
+        CountDownLatch protobufDescriptorTxLatch = new CountDownLatch(numExpectedTxOnProtobufDescriptorTable);
+        StreamingUpdateListener protobufDescriptorTxListener = new StreamingUpdateListener(protobufDescriptorTxLatch);
+        corfuStoreStandby.subscribeListener(protobufDescriptorTxListener, TableRegistry.CORFU_SYSTEM_NAMESPACE,
+            ObjectsView.getLogReplicatorStreamId().toString(),
+            Arrays.asList(TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME));
+
         CountDownLatch streamingUpdatesLatch =
             new CountDownLatch(totalStreamingUpdates);
         StreamingUpdateListener streamingUpdateListener =
             new StreamingUpdateListener(streamingUpdatesLatch);
         corfuStoreStandby.subscribeListener(streamingUpdateListener, NAMESPACE, TAG_ONE);
 
-        startLogReplicatorServersWithCustomMaxWriteSize(maxWriteSize);
+        // Start LR on both clusters with custom write sizes.
+        startLogReplicatorServersWithCustomMaxWriteSize();
 
         log.debug("Wait for snapshot sync to finish");
         statusUpdateLatch.await();
         streamingUpdatesLatch.await();
+        protobufDescriptorTxLatch.await();
```

Comment:
I think the count down on `numExpectedTxOnProtobufDescriptorTable` doesn't necessary means all entries of ProfobufDescriptorTable replicated to the Sink right? as the StreamingTask for this table is tracking the lr transaction stream, then all the LR replicated entries will count down this latch?

Reply:
I missed some parts in streaming task creation, addressed offline